root/static/scripts/guess-case/modes.js

```diff
@@ -17,8 +17,9 @@ const {isPrepBracketWord, isPrepBracketSingleWord} = require('./utils');
  * keschte  2005-04-17  added french lowercase characters
  * keschte  2005-06-14  added "tha" to be handled like "the"
  * warp     2011-02-01  added da, de, di, fe, fi, ina, inna
+ * dpmittal 2016-12-20  added turkish lowercase characters
  */
-const LOWER_CASE_WORDS = /^(a|an|and|as|at|but|by|da|de|di|fe|fi|for|in|ina|inna|n|nor|o|of|on|or|tha|the|to)$/;
+const LOWER_CASE_WORDS = /^(a|an|and|as|at|but|by|da|de|di|fe|fi|for|in|ina|inna|n|nor|o|of|on|or|tha|the|to|ve|ile|ya|veya|yahut|ki|mı|mi|mu|mü|mısın|musun|mudur|mıdır|midir|miyim|misin|misiniz|mısınız|müsün|müyüm|müsünüz)$/;
```

Comment:
Again, sorry for responding slow. 

Here is the short response to all variants of the question word: 

"mı|mi|mu|mü|mıyım|miyim|muyum|müyüm|mısın|misin|musun|müsün|mıyız|miyiz|muyuz|müyüz|mısınız|misiniz|musunuz|müsünüz|mıdır|midir|mudur|müdür"

"müdür" is a problematic case as @uklauer mentioned. While it is almost always easy to distinguish the usage from the context, it would not be straightforward for a machine without incorporating some natural language processing. 

Below I am adding detailed explanations, in case it's needed. If you need help with other suffixes, conjuctions etc., please don't hesitate to ask. My apologies that I might not reply immediately for the next  days.

----------------------

The base question word in Turkish is "mı" with its conjugations "mi, mu, mü." According to the 	subject (tr. özne) of the sentence, a suffix is added. I'm adding a table of all variants of the question word below:

| Sample Verb         | yapar          | gelir          | uyur          | ölür          |
|---------------------|----------------|----------------|---------------|---------------|
| __1st Person Singular__ | mıyım          | miyim          | muyum         | müyüm         |
| __2nd Person Singular__ | mısın          | misin          | musun         | müsün         |
| __3rd Person Singular__ | mı             | mi             | mu            | mü            |
| __1st Person Plural__   | mıyız          | miyiz          | muyuz         | müyüz         |
| __2nd Person Plural__   | mısınız        | misiniz        | musunuz       | müsünüz       |
| __3rd Person Plural__   | (yapar-lar) mı | (gelir-ler) mi | ~~(uyur-lar) mı~~ | ~~(ölür-ler) mi~~ |

Grammatically the third person plural can only be "mı" or "mi" because of the plural suffix "ler" or "lar" added to the end of the verb.

If there is "no verb," i.e. we use the form which corresponds to "to be" in English, the subject is the third person (singular or plural). Hence, the question word could only be "mı, mi, mu, mü." In this case one can optionally add "dır" (or its harmonic variants "dir, dur, dür") for emphasis.

| Sample Noun         | yalan          | kel          | balon          | düş          |
|---------------------|----------------|----------------|---------------|---------------|
| __* to be__             | mı(dır)          | mi(dir)          | mu(dur)         | mü(dür)         |

The above case also holds for several tenses, e.g. "yapacak mıdır"